texar/torch/modules/classifiers/conv_classifiers.py

```diff
@@ -69,7 +69,13 @@ def __init__(self, in_channels: int, in_features: Optional[int] = None,
         self._num_classes = self._hparams.num_classes
         if self._num_classes > 0:
             if self._hparams.num_dense_layers <= 0:
+                if in_features is None:
+                    raise ValueError("'in_features' is required for logits "
+                                     "layer when 'num_dense_layers' <= 0")
                 self._encoder.append_layer({"type": "Flatten"})
+                ones = torch.ones(1, in_channels, in_features)
+                input_size = self._encoder._infer_dense_layer_input_size(ones)  # pylint: disable=protected-access
```

Comment:
Now that we have `output_size`, do we still need this?

Reply:
I am not sure I understand the logic of `output_size` method. I think this was added after the PR for Convolution Networks was merged. It returns `self._hparams.out_channels * len(self._hparams.kernel_size)` which is wrong. First `self._hparams.out_channels` can be a list. Even if it is not a list, when we have more than one convolution layers (For e.g., a merge layer followed by a single convolution layer), the above returned value is wrong. One way to know the output size is to pass the dummy tensor through the network, another way is to work it out through a mathematical formula.

Reply:
I see. Could you try to correct that implementation? I would prefer we work out a formula to compute the size, rather than having to do a forward pass.

Reply:
The formula can be a bit tricky to write, as it has to consider several cases of MergeLayer combinations, stride sizes, kernel sizes of several layers. Can we do it as part of a separate PR? A dummy pass is a simple solution. And regarding the output size, we actually cannot know it unless we know the size of input tensor. That is the reason we take `in_channels` and `in_features` as part of arguments in the constructor.

Reply:
We could do it as a separate PR, but I think it's something we should have. 
If there are no dense layers then I think the output size should just be the output channel count (at least in NLP this makes sense), and we should be able to know that given `in_channels` alone (there's actually a formula on PyTorch docs).
If dense layers exist, we'll just use the size of the output of the dense layer.

Reply:
Added a fix in fb85495